pkg/constants/constants.go

```diff
@@ -128,20 +127,6 @@ func GetDefaultSquashfsCompressionOptions() []string {
 	return []string{"-comp", "gzip"}
 }
 
-func GetGrubFilePaths(arch string) []string {
```

Comment:
I'm concerned on the "legacy" portion of Kairos, but most importantly this mechanism was used also to protect from nuances of depending on the GRUB version shipped from the distro. It gives more control in what are you actually using.

I'm not against this change, but I'm saying we should fallback if we don't find an EFI file in the expected locations - so to say, let's have a `/usr/share/kairos/efi/*` if that's present takes precedence, otherwise we scan and if we don't find it, we just fail. That is also a good safety net meanwhile switching to use native packages wherever needed.

Reply:
we have a fallback efi already but this is done when building the iso: https://github.com/kairos-io/enki/blob/main/pkg/action/build-iso.go#L267 and https://github.com/kairos-io/enki/blob/main/pkg/action/build-iso.go#L278

if we dont find efi files in the expected locations, we fall back to the generic one provided by the packages.

Reply:
if we want secureboot this is the only way of providing it, by using artifacts directly from the distro owners. Its similar to the kernel part, we dont know the difference in kernels and that can screw us easily, but...we expect the upstream distros to have done their due diligence and make sure it works.

Otherwise there is no way of supporting secureboot unless we use our own keys to sign everything (including the kernel!)

Reply:
> gotcha, I see, maybe we can start to make this more first-class in Kairos and have our own location (like we do for /etc/os-release variables for instance) easier to track and we don't have to overwrite distro-specific paths (that might be hard to realize in case of debug, totally not-obvious)

yes, this was my idea as well, provide them in a very specific place instead of all along, but this change has already so many changes across so many places that I didnt wanted to introduce yet another change to the packages without having this out and properly tested (the idea is to add a couple of tests in kairos to use EFI+secureboot once this is in)